### PR TITLE
Initial rough arrow documentation

### PR DESCRIPTION
**Pull request by blaxill**

### CODE REVIEW

File: cava/doc/arrows.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+## Arrow usage in this project
+
+This project currently uses the generalized arrows described by Adam Megacz
+work with some notable differences:
+
+- Currently the Arrow sub classes are not broken out into their own classes (e.g
+    ArrowDrop)
+- Arrows are used "directly" and we don't represent lanuages as generalized
+    arrows. Adams construction of multilevel languages makes heavy use of the
```
Comment (satnam6502) [inline cava/doc/arrows.md:68]: Adam's

File: cava/doc/arrows.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    instance:
+
+      - Evaluation equivalence is pointwise equivalence (e.g. `f~g if f(x)~g(x)`)
+      - The constructive instance equivalence is structural equivalence with
+          some allowed transforms such as `f >>> drop ~ drop`, that is, dropping
+          all outputs from some component is equivalent to immediately dropping.
+      - Netlist equivalence is not implemented but should perhaps be if the hdl is
+          equivalent up to reordering.
+
+Kappa Calculus is encoded as an inductive type using the PHOAS method, with an
```
Comment (satnam6502) [inline cava/doc/arrows.md:86]: Link to original PHOAS paper?

File: cava/doc/arrows.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+Inductive kappa : object -> object -> Type :=
+    | Var : forall x y,   var x y -> kappa x y
+    | Abs : forall x y z, (var unit x -> kappa y z) -> kappa (x**y) z
+    | App : forall x y z, kappa (x**y) z -> kappa unit x -> kappa y z
+    | Compose : forall x y z, kappa y z -> kappa x y -> kappa x z
+    | Arr : forall x y,   morphism x y -> kappa x y.
+```
+
+Terms can be constructed directly but an experimental notation is also provided.
+
```
Comment (satnam6502) [inline cava/doc/arrows.md:99]: How about a few very simple examples directly in `kappa`? A nand gate? Something with a state element and feedback?  And something with a forked wire?

File: cava/doc/arrows.md (hunk 1 above)
Comment (blaxill) [reply cava/doc/arrows.md]: Done thanks

File: cava/doc/arrows.md (hunk 3 above)
Comment (blaxill) [reply cava/doc/arrows.md:150]: Updated examples

File: cava/doc/arrows.md (hunk 2 above)
Comment (blaxill) [reply cava/doc/arrows.md]: Done

**Pull request merged**